test/serviceprofiles/serviceprofiles_test.go

```diff
@@ -117,7 +134,135 @@ func TestServiceProfiles(t *testing.T) {
+func TestServiceProfileMetrics(t *testing.T) {
+
+	testNamespace := TestHelper.GetTestNamespace("serviceprofile-test")
+	testCases := []string{
+		"retries",
+		"latency",
+	}
+
+	for _, tc := range testCases {
+		var (
+			tc                   = tc
+			testSP               = fmt.Sprintf("world-%s-svc", tc)
+			testDownstreamDeploy = fmt.Sprintf("deployment/world-%s", tc)
+			testUpstreamDeploy   = fmt.Sprintf("deployment/hello-%s", tc)
+			testYAML             = fmt.Sprintf("testdata/hello_world_%s.yaml", tc)
+		)
+
+		t.Run(tc, func(t *testing.T) {
+			out, stderr, err := TestHelper.LinkerdRun("inject", testYAML)
+			if err != nil {
+				t.Errorf("'linkerd %s' command failed with %s: %s\n", "inject", err.Error(), stderr)
+			}
+
+			out, err = TestHelper.KubectlApply(out, testNamespace)
+			if err != nil {
+				t.Errorf("kubectl apply command failed\n%s", out)
+			}
+
+			assertion := &routeStatAssertion{
+				upstream:   testUpstreamDeploy,
+				downstream: testDownstreamDeploy,
+				namespace:  testNamespace,
+			}
+
+			cmd := []string{
+				"profile",
+				"--namespace",
+				testNamespace,
+				"--open-api",
+				"testdata/world.swagger",
+				testSP,
+			}
+
+			out, stderr, err = TestHelper.LinkerdRun(cmd...)
+			if err != nil {
+				t.Errorf("'linkerd %s' command failed with %s: %s\n", cmd, err.Error(), stderr)
+			}
+
+			_, err = TestHelper.KubectlApply(out, testNamespace)
+			if err != nil {
+				t.Errorf("kubectl apply command failed:\n%s", err)
+			}
+
+			assertRouteStat(assertion, t, func(stat *rowStat) error {
+				if stat.EffectiveSuccess != stat.ActualSuccess {
+					return fmt.Errorf(
+						"expected Effective Success to be equal to Actual Success but got: Effective [%.2f] <> Actual [%.2f]",
+						stat.EffectiveSuccess, stat.ActualSuccess)
+				}
+				return nil
+			})
+
+			profile := &sp.ServiceProfile{}
+
+			// Grab the output and convert it to a service profile object for modification
+			err = yaml.Unmarshal([]byte(out), profile)
+			if err != nil {
+				t.Errorf("unable to unmarshall YAML: %s", err.Error())
+			}
+
+			for _, route := range profile.Spec.Routes {
+				if route.Name == "GET /testpath" {
+					route.IsRetryable = true
+					break
+				}
+			}
+
+			bytes, err := yaml.Marshal(profile)
+			if err != nil {
+				t.Errorf("error marshalling service profile: %s", bytes)
+			}
+
+			out, err = TestHelper.KubectlApply(string(bytes), testNamespace)
+			if err != nil {
+				t.Errorf("kubectl apply command failed:\n%s :%s", err, out)
+			}
+
+			assertRouteStat(assertion, t, func(stat *rowStat) error {
+				if stat.EffectiveSuccess <= stat.ActualSuccess {
+					return fmt.Errorf(
+						"expected Effective Success to be greater than Actual Success but got: Effective [%f] <> Actual [%f]",
+						stat.EffectiveSuccess, stat.ActualSuccess)
+				}
+				return nil
+			})
+		})
+	}
+}
+
+func assertRouteStat(assertion *routeStatAssertion, t *testing.T, assertFn func(stat *rowStat) error) {
```

Comment:
tioli: since `routeStatAssertion` is only used in one place, i'd probably just pass `upstream`, `downstream`, and `namespace` directly into `assertRouteStat`